src/sys/inotify.rs

```diff
@@ -145,50 +153,47 @@ pub struct InotifyEvent {
     pub cookie: u32,
     /// Filename. This field exists only if the event was triggered for a file
     /// inside the watched directory.
-    pub name: OsString
+    pub name: Option<CString>
```

Comment:
I'm a little hesitant about putting `CString` in a public API.  `CString` is harder to work with than other types, like `String`, `OsString`, or even `PathBuf`.

Reply:
I agree with that. However no other type fits particularly well. I could maybe return a `String` using `CStr::to_string_lossy` but we would lose data. There is no easy path from `CString` to `OsString` and I'm not sure `PathBuf` is the right structure to use.